src/components/Auth/InputCustom.tsx

```diff
@@ -0,0 +1,94 @@
+export interface InputCustomProps {
+  formValue: string
+  label: string
+  placeholder?: string
+  type?: string
+  required?: boolean
+  validation?: any
+  messageError?: string
+}
+import {
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  Icon,
+  Input,
+  InputGroup,
+  InputRightElement,
+  Text,
+} from '@chakra-ui/react'
+import { useState } from 'react'
+import { useFormContext } from 'react-hook-form'
+import { useTranslation } from 'react-i18next'
+import { MdOutlineRemoveRedEye } from 'react-icons/md'
+import { RiEyeCloseLine } from 'react-icons/ri'
+import useDarkMode from '~src/themes/saas/hooks/useDarkMode'
+
+export interface InputCustomProps {
+  formValue: string
+  label: string
+  placeholder?: string
+  type?: string
+  required?: boolean
+  validation?: any
+  messageError?: string
+}
+
+const InputCustom = ({
+  formValue,
+  label,
+  placeholder,
+  type = 'text',
+  required = false,
+  validation = {},
+}: InputCustomProps) => {
+  const { t } = useTranslation()
+  const { textColor, textColorBrand, textColorSecondary } = useDarkMode()
+  const [show, setShow] = useState(false)
+  const handleClick = () => setShow(!show)
+
+  const {
+    register,
+    formState: { errors },
+  } = useFormContext()
+
+  const validationRules = {
+    ...validation,
+    ...(required ? { required: { value: true, message: t('form.error.field_is_required') } } : {}),
+  }
+
+  const errorMessage = errors[formValue]?.message?.toString() || ''
+
+  const inputType = type === 'password' && !show ? 'password' : type === 'password' ? 'text' : type
```

Comment:
This is better to be something like

```ts
let inputType = type

if(type ===  'password' && show) {
     inputType = text
}
```

Much clear :) 

Try to do not use ternary operators unless is the only way.